nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java

```diff
@@ -168,17 +216,49 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
         final long now = System.currentTimeMillis();
 
         final ComponentLog logger = getLogger();
+        final String monitoringScope = context.getProperty(MONITORING_SCOPE).getValue();
+        final boolean copyAttributes = context.getProperty(COPY_ATTRIBUTES).asBoolean();
+        final boolean isClusterScope = SCOPE_CLUSTER.equals(monitoringScope);
         final List<FlowFile> flowFiles = session.get(50);
+
+        boolean isInactive = false;
+        long updatedLatestSuccessTransfer = -1;
+        StateMap clusterState = null;
+        final boolean shouldThisNodeReport = !isClusterScope
```

Comment:
Good call. I'll move this condition to a function and call it just before reporting attempts.